RationaleMCP/0039/SEMLA.md

```diff
@@ -0,0 +1,432 @@
+installs from the archive extract either only the “.library” folder, or
+the entire contents of the directory tree. When extracting the entire
+contents of the directory tree, The directory structure and file names
+of the content of the library directory must be the same as in the
+archive, except that the top-level directory may be placed wherever and
+named however the tool decides. It is recommended that the directory is
+placed in a directory on the MODELICAPATH and named “*PACKAGENAME*” or
+“*PACKAGENAME VERSION*”, as per the Modelica Language Specification.
+
+## Reading open-source library
+
+Reading a library without encryption is done in the same way as a
+library stored on disk as described by the Modelica Language
+Specification.
+
+## Reading an encrypted library
+
+1.  Read the XML file with metadata (“manifest.xml”) from the “.library”
+    directory, specifically the path to the LVE suitable for the current
+    platform.
+
+2.  Start LVE.
+
+3.  Communicate with LVE through its stdin & stdout, according to the
+    protocol described below.
+
+4.  Modelica files with a “.moc” file extension are read through the LVE
+    instead of from disk. Other files in the library may also be read
+    through the LVE, if desired.
+
+## Communication Protocol
+
+### Overview
+
+This protocol is intended to be used for proprietary libraries.
+Open-source libraries are stored as plain text within the library
+archive. The communication protocol is through stdin and stdout of the
+LVE and uses encryption of the communication channel to prevent
+eavesdropping. For this encryption, both the tool and the LVE need a
+public-private key pair.
+
+The protocol has three phases. The first phase is a TLS (Transport Layer
+Security, the successor of SSL) cryptographic handshake that sets up a
+secure communication between the tool and the LVE. In the second phase,
+what version of the protocol to use is established. The third phase is
+decryption of Modelica code and/or licensing. Note that the encryption
+used in the communication is entirely separate from the encryption used
+for storing the library.
+
+For authentication, the LVE contains a list of the public keys of the
+tools that it trusts. The tool implicitly trusts that the LVE has
+permission to license the library because it can decrypt the library.
+
+The picture below shows the parties involved in the communication and
+the keys used. The LVE reads the encrypted files in the library archive
+file, decrypts them, and sends them through the encrypted communication
+tunnel to the tool.
+
+<img src="keys_overview.png" style="width:6.39931in;height:1.72083in" />
+
+### Communication flow
+
+The picture below details the flow of the communication between the tool
+and the LVE.
+
+<img src="Protocol_flow.png" style="width:6.39931in;height:4.73373in" />
+
+### Handshake
+
+1.  The tool initiates a cryptographic handshake according to TLS 1.2
```

Comment:
TLS 1.2 has been superseded by TLS 1.3; rfc 8446. I assume that should be used, right?

Reply:
FWIW: I am getting a lot of requests from Bosch and others to move this MCP forward. @HansOlsson, andy chance to put it on the discussion agenda? I think it would be good to get synchronized feedback on the status.